src/sysv/systemd/set-nis-domainname.in

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/sh
```

Comment:
Hi,

why is the step with the shell script needed. Wouldn't itbe more straight forward to read `/etc/sysconfig/network` in `sssd.service` and call `setdomainname` directly with `ExecStartPre` ?

bye,
Sumit

Reply:
@sumit-bose please see https://github.com/SSSD/sssd/pull/7674#issuecomment-2451433107

Reply:
> Wouldn't itbe more straight forward to read `/etc/sysconfig/network` in `sssd.service` and call `setdomainname` directly with `ExecStartPre` ?

I considered this: feels like a simple `grep` + `cut` would do the trick of extracting `NISDOMAIN` from `/etc/sysconfig/network`

But then I recalled my previous attempt to embed a meaningful shell script into service file - #7607 - and decided to not even try this time :)

FWIW, if script will grow more complicated than `grep` + `cut`, standalone file will be easier to edit than embedded one-liner...

Reply:
@abbra, thanks for the hint, I will read existing comments more carefully next time.

@alexey-tikhonov, if I understand to correctly `ExecStartPre=+-@libexecdir@/sssd/set-nis-domainname` means we run a script (`/etc/sysconfig/network`) of unknown source with full privileges? Would it be possible to lower the privileges either in the systemd service file or in the `set-nis-domainname`?

bye,
Sumit

Reply:
We need to run `set-nis-domainname` in forked process with fully stripped privs.

Reply:
'll try to fix this.

Reply:
`/etc/sysconfig/network` has to be treated as trusted. It is controlled by the root anyway.

Reply:
> Would it be possible to lower the privileges either in the systemd service file or in the `set-nis-domainname`?

@sumit-bose, please take a look at my take.

It's based on `capsh` tool:
```
# capsh --print
Current: =ep
Bounding set =cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,cap_wake_alarm,cap_block_suspend,cap_audit_read,cap_perfmon,cap_bpf,cap_checkpoint_restore

# capsh --mode=NOPRIV -- -c "capsh --print"
Current: =
Bounding set =
...
Securebits: 0357/0xef/8'b11101111 (no-new-privs=1)
 secure-noroot: yes (locked)
 secure-no-suid-fixup: yes (locked)
 secure-keep-caps: no (locked)
 secure-no-ambient-raise: yes (locked)

# capsh --inh=cap_sys_admin --drop=all -- -c "capsh --print"
Current: cap_sys_admin=eip
Bounding set =
```

Ironically, now `set-nis-domainname` script requires **_more_** privileges to execute: in addition to `CAP_SYS_ADMIN` required by `getdomainname()`, it now also needs `CAP_SETPCAP` to be able to manipulate own secure bits (--mode=) and bounding set (--drop=)
But it's run fully privileged anyway (is there a simple way to specify capabilities for a specific `ExecStartPre`..?)

But at least `/etc/sysconfig/network` should be now run fully unprivileged and `setdomainname` with single `CAP_SYS_ADMIN`

As another option, I tried to find a way to split this into two `ExecStartPre`  --  one unprivileged to run `/etc/sysconfig/network` and export `NISDOMAIN` and another one privileged to run `setdomainname` directly, but didn't find a straightforward way (besides writing an intermediate env file).

Another option would be to get rid of script altogether and instead pipe things right inside service file, something like:
```
ExecStartPre=+-/bin/sh -c "capsh --mode=NOPRIV -- -c "source $NETWORK; echo \$NISDOMAIN" | xargs setdomainname"
```
but real code will require a lot of escaping and will get unreadable...